webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/services/FileManagerJaxrsServiceV2.java

```diff
@@ -0,0 +1,235 @@
+    }
+
+
+    /**
+     * Gets the file manager's working directory from the servlet context.
+     *
+     * @return the file manager working directory
+     * @throws Exception if an object cannot be retrieved from the context
+     *                   attribute
+     */
+    public File getContextWorkingDir() throws CasProductException {
+        Object workingDirObject = context.getAttribute("workingDir");
+        if (workingDirObject != null && workingDirObject instanceof File) {
+            return (File) workingDirObject;
+        }
+
+        String message = ErrorTypes.CAS_PRODUCT_EXCEPTION_FILEMGR_WORKING_DIR_UNAVILABLE.getErrorType();
+        LOGGER.log(Level.WARNING, message);
+        throw new CasProductException(message);
+    }
+
+
+    /**
+     * Gets the file manager client instance from the servlet context.
+     *
+     * @return the file manager client instance from the servlet context attribute
+     * @throws Exception if an object cannot be retrieved from the context
+     *                   attribute
+     */
+    public FileManagerClient getContextClient()
+            throws CasProductException {
+        // Get the file manager client from the servlet context.
+        Object clientObject = context.getAttribute("client");
+        if (clientObject != null &&
+                clientObject instanceof FileManagerClient) {
+            return (FileManagerClient) clientObject;
+        }
+
+        String message = ErrorTypes.CAS_PRODUCT_EXCEPTION_FILEMGR_CLIENT_UNAVILABLE.getErrorType();
+        LOGGER.log(Level.WARNING, message);
+        throw new CasProductException(message);
+    }
+
+
+
+
+
+    @GET
+    @Path("product")
+    @Produces({"application/xml", "application/json", "application/atom+xml",
+            "application/rdf+xml", "application/rss+xml", "application/zip"})
+    public ProductResource getProduct(@QueryParam("productId") String productId) throws WebApplicationException
+    {
+        if (productId == null || productId.trim().equals(""))
+        {
+            throw new BadRequestException(ErrorTypes.BAD_REQUEST_EXCEPTION_PRODUCT_RESOURCE.getErrorType());
+        }
+
+        try
+        {
+            FileManagerClient client = getContextClient();
+
+            // Find the product.
+            Product product = client.getProductById(productId);
+            product.setProductReferences(client.getProductReferences(product));
+
+            // Create the product resource, add the product data and return the
+            // resource as the HTTP response.
+            return new ProductResource(product, client.getMetadata(product),
+                    product.getProductReferences(), getContextWorkingDir());
+        }
+        catch (Exception e)
+        {
+            // Just for Logging Purposes
+            String message = "Unable to find the requested resource.";
+            LOGGER.log(Level.FINE, message, e);
```

Comment:
It will be ok to print an error log. Fine for now.

Reply:
Logs used for just testing purposes. Exceptions are thrown as payloads.